### DISCUSSION TITLE
database names should be singular

### DISCUSSION THREAD

**Discussion by baumannalexj** [strapi/strapi#6081] [Legacy Topics]
We found strapi for a work project, and we love the idea! Our data flow has strict naming conventions: We believe table names should be singular as the correspond to a single row entity. 

Strapi should allow table names to be singular.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by baumannalexj** [reply]
To clarify "Strapi should allow table names to be singular" is the request.

A config for DB entity naming convention should be available to comply with user db standards 
e.g.:
 * entities to have snake_case_names = true
 * entities names are singular = true
...

**Reply by derrickmehaffy** [reply]
Please see the `collectionName` setting in the model settings: https://strapi.io/documentation/3.0.0-beta.x/concepts/models.html#model-settings

**Comment by robertpiosik** [comment]
[Postgres](https://www.postgresql.org/docs/current/ddl-basics.html) in their docs creates table with a plural name, [MySQL](https://dev.mysql.com/doc/refman/8.0/en/creating-tables.html) otherwise. So I won't say that it should be done in one way or another.

For me personally table represents a collection of rows, not a single row. So I would name it using a plural name, as it is done by default.

**Reply by derrickmehaffy** [reply]
I feel the same way myself but :shrug:

**Reply by baumannalexj** [reply]
Thanks for following up on this.

Right I should have quoted that part to show it's our Organization's and Data team's standards:
> Our data flow has strict naming conventions: "We believe table names should be singular as the correspond to a single row entity."

Interestingly -- we just found that when you create a collection with "advance settings" and use a singular name -- the table is stored as singular.

Is it intended that naming a collection has a different standard when created via the regular flow?

**Reply by robertpiosik** [reply]
This is simply a way to name the table however you want :)

**Reply by baumannalexj** [reply]
Oh ha, so this is exactly what we want then :]

Thanks @robertpiosik .

**Comment by goxiaoy** [comment]
So why the new created content type collection name design to be plural?
It causes so many problems to handle auto transform singular to plural.
Why not just keep what exactly user types

And I notice that plugins in this core repository keep singular table names. like `users-permissions_user`, `upload_file`

**Reply by derrickmehaffy** [reply]
I will ping @alexandrebodin and @petersg83 about this

**Reply by petersg83** [reply]
We know the trouble, we feel it too :/ The singular/plural system is a legacy that will change in a future major release of strapi but we have no ETA for the moment
